beacon-chain/forkchoice/doubly-linked-tree/types.go

```diff
@@ -11,12 +11,12 @@ import (
 
 // ForkChoice defines the overall fork choice store which includes all block nodes, validator's latest votes and balances.
 type ForkChoice struct {
+	lock                sync.RWMutex // global forkchoice lock
```

Comment:
If you do this then you do not need to redefine the lock and unlock methods

```suggestion
	sync.RWMutex // global forkchoice lock
```